gh-120291: Fix a bashism in python-config.sh.in (#120292)
python/cpython: Misc/NEWS.d/next/Build/2024-06-09-15-54-22.gh-issue-120291.IpfHzE.rst

> Fix the shell version of ``python-config`` script to work with non-bash
> shells.

Reviewer: ```suggestion
Make the ``python-config`` shell script compatible with non-bash shells.
```

Strictly speaking, POSIX does not require the `local` keyword to be valid in scoped functions. So, `exit_with_usage` is still not fully POSIX compliant. If you want a purely POSIX compliant script, you should use sub-shells (but that's an overkill and most, if not all, implementations of POSIX support `local`).

Author: Updated, thanks. Also split the `local` while at it, that should get us the best of both worlds. Even if `local` isn't supported by the shell, the assignment will still work.

Reviewer 2: It would also print a pointless error message to the console.

I don't actually understand the point of using `local` here at all though, since it happens immediately before exiting the script entirely. Why does it need to set up a separate variable context to avoid clobbering global shell variables?

Reviewer: @mgorny Actually, I just understood the "Keep this script in sync with python-config.in". There's a copy of that script in `Misc` so you should also fix that part.

@eli-schwartz You're right. There is also the RESULT variable which appears to be global and AFAICT, the latter is not used outside that script (nor is there any pollution from other scripts or the Makefile but I may be wrong).

Author: > @mgorny Actually, I just understood the "Keep this script in sync with python-config.in". There's a copy of that script in `Misc` so you should also fix that part.

This in `Misc`. There are two scripts: `python-config.sh.in` that's written in shell and I'm fixing that one, and `python-config.in` that's written in Python and isn't affected.
